Add granite training example

mtake opened the pull request:
This PR is to add training examples for granite-3.3-8b-instruct model.
granite-3.3-8b-instruct is the student model for Japanese and we have been using the code for training the model.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * CLI tool to interpolate between two compatible model checkpoints and save the result.
  * Single-node multi-GPU example workflows for SFT and OSFT on Granite 3.3 8B Instruct with configurable training, checkpointing, scheduler, and optional post-training interpolation.
  * Automatic discovery of latest checkpoints and conditional interpolation via a weight parameter.
  * Improved run outputs: timing, completion summaries, and OOM mitigation guidance.

* **Documentation**
  * Added an experimental "Model Interpolation" guide and example entries for SFT/OSFT (also referenced in memory estimation).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Maxusmusti commented on examples/scripts/interpolator.py:
```diff
@@ -0,0 +1,135 @@
+    torch_dtype: str | torch.dtype | None = "bfloat16",
+) -> str:
+    if output_model_path is None:
+        output_model_path = f"{trained_model_path}_interp"
```
It would be nice if this script printed something along the lines of "Merged model saved at <output_model_path" once saved. That way it quickly indicates success and people know where to find result when running with default settings, without looking at code

Maxusmusti reviewed (commented):
@mtake Thanks again for the contribution! Everything looks good, just a few minor nits, if you have time might be worth addressing, but if not can be merged as-is. I tested the scripts and all seems to be working as expected.

mtake commented:
Hi @Maxusmusti 
I addressed all your comments in 2cd1d2a9ca98790a1a7a357eaac4829f7ce27faf. The comments were useful to make the code comprehensive. Thank you!
